src/System.Device.Gpio/System/Device/Pwm/PwmChannel.cs

```diff
@@ -67,14 +67,27 @@ protected virtual void Dispose(bool disposing)
                     frequency,
                     dutyCyclePercentage);
             }
-            else
+
+            if (IsBeagleBoneKernel())
```

Comment:
NIT: instead of two if blocks, can we do if()... else if ()... else?

It is just the convention we usually follow.